dask/distributed/tests/test_ipython_utils.py

```diff
@@ -0,0 +1,68 @@
+import time
+from subprocess import Popen, PIPE
+
+import pytest
+
+np = pytest.importorskip('numpy')
+pytest.importorskip('IPython.parallel')
+
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+
+from IPython.parallel import Client
+
+from dask.distributed import dask_client_from_ipclient
+import dask.array as da
+
+
+def setup_module():
+    start_cmd = ['ipcluster', 'start', '-n', '4']
+    cp = Popen(start_cmd, stdout=PIPE, stderr=PIPE)
+    tic = time.time()
+    while True:
+        line = cp.stderr.readline().decode()
+        if 'Engines appear to have started' in line:
+            break
+        elif 'CRITICAL | Cluster is already running' in line:
+            raise RuntimeWarning("Cluster is already running")
+        elif time.time() - tic > 50:
+            raise RuntimeError("Timeout waiting for cluster to start.")
+        time.sleep(0.1)
+
+
+def teardown_module():
+    kill_cmd = ['ipcluster', 'stop']
+    cp = Popen(kill_cmd, stdout=PIPE, stderr=PIPE)
+    tic = time.time()
+    while True:
+        line = cp.stderr.readline().decode()
+        if "Stopping cluster" in line:
+            break
+        elif time.time()-tic > 15:
+            # probably stopped or something went wrong
+            break
+        time.sleep(0.1)
```

Comment:
I would prefer using their TestCases.  Some reasons:

1.  We might want more than one test in this module with fresh ipclusters
2.  I suspect that this solution, while neat and simple, doesn't catch all of the strange things that might happen with ipcluster and ZMQ
3.  I am ignorant of how to troubleshoot and solve ipcluster issues that might arise in future tests should this solution not be comprehensive.
4.  I strongly desire to stay ignorant of those issues

And so my approach would be to install `nose` in .travis.yml and build off of their `ClusterTestCase`.  I'm happy to manage that if you're feeling done with this PR.  Also, disagreements heartily welcome; you've had more experience here than I have.